README.md

```diff
@@ -187,3 +183,126 @@ After that you can start QEMU
+    - Content: Download and deploy Intel ACM `*.bin` files.
+* skl
+    - Recipe: recipes-support/skl/skl_git.bb
+    - Content: Secure Kernel Loader
+* linux-tb
+    - Recipe: recipes-kernel/linux/linux-tb_6.6.bb
+    - Content: Linux kernel
+* grub
+    - Recipe: recipes-bsp/grub/grub_%.bbappend
+* grub-efi
+    - Recipe: recipes-bsp/grub/grub-efi_%.bbappend
+* grub & grub-efi
+    - Recipe: recipes-bsp/grub/grub-tb-common.inc
+    - Content: Common config for both recipes
+
+### Source revision
+
+To change branch or commit used by a recipe you have to change `BRANCH` or
+`SRCREV` variable in appropriate recipe file. In case of Linux kernel those
+variables are named `KBRANCH` and `SRCREV_machine`
+
+### Building modified source
+
+In order to make and test changes to source code you first need to enter kas
+shell
+
+```shell
+kas-container shell meta-trenchboot/kas-generic-tb.yml
+```
+
+After that use `devtool modify <recipe>` to fetch source code of the recipe you
+want to work on. Bitbake will checkout branch and commit set by `BRANCH` and `SRCREV`
+variables.
+
+```shell
+devtool modify skl
+(...)
+INFO: Source tree extracted to /build/workspace/sources/skl
+INFO: Using source tree as build directory since that would be the default for this recipe
+INFO: Recipe skl now set up to build from /build/workspace/sources/skl
+```
+
+All recipes' sources you wish to modify will be in `/build/workspace/sources`.
+After modifications, you can try to a build recipe by using
+`devtool build <recipe>` or if you want to build the whole image with your changes
+then use `devtool build-image tb-minimal-image`.
+After building the image, you can [install](#flash) and [boot](#booting) it or
+run it in [QEMU](#running-in-qemu).
+
+#### GRUB
+
+Currently, if you try to build `grub` or `grub-efi` twice you will get error
+
+```text
+| Updating file build-aux/config.rpath (backup in build-aux/config.rpath~)
+| Updating file m4/extern-inline.m4 (backup in m4/extern-inline.m4~)
+| gnulib/gnulib-tool: *** file /build/workspace/sources/grub/gnulib/m4/lib-ld.m4 not found
+| gnulib/gnulib-tool: *** Stop.
+| /build/workspace/sources/grub/bootstrap: gnulib-tool failed
+| WARNING: exit code 1 from a shell command.
+```
+
+To fix this you have to delete `/build/workspace/sources/<recipe>/gnulib`
+directory before building recipe again.
+
+### Local files
+
+Files added by Yocto recipe are stored inside `sources/<recipe>/oe-local-files`
+folder. Example of local file is `defconfig` file in `linux-tb` recipe
+
+### Deployment
+
+To deploy component to target machine after making changes you can use rsync
+with path depending on component you modified. `<destination>` can be either
+`<USER>@<IP>:/` or path to mounted rootfs partition. Similarly,
+`<boot_destination>` is `<USER>@<IP>:/boot` or path to mounted boot partition.
+To deploy files via ssh you need to start/enable ssh server on target machine.
+
+* grub
```

Comment:
@m-iwanicki  Was it tested? I am afraid that legacy grub is a bit more complex.
Legacy grub gets installed directly on the disk.

We use grub-mkimage and grub-bios-setup commands here to install the bootloader on the **image file**: https://github.com/zarhus/poky/blob/wic-plugin-pcbios-grub/scripts/lib/wic/plugins/source/bootimg-pcbios.py#L117

Have you looked through: https://github.com/TrenchBoot/trenchboot-sdk/blob/master/scripts/aem/aem.sh ?

It is probably this command: https://github.com/TrenchBoot/trenchboot-sdk/blob/master/scripts/aem/aem.sh#L238 that installs grub here at runtime

Do you want to add something here @SergiiDmytruk @krystian-hebel ?

Reply:
Yes, this is the command. This doesn't have to be performed on the target, I installed GRUB locally on pendrive when testing Xen SMP changes earlier from the host, but I believe I had to also pass a source location of modules.

Reply:
Managed to do legacy grub deployment but it's still not 1 to 1 in https://github.com/3mdeb/meta-trenchboot/pull/29/commits/5c558b59998af224856be23d8aa3d07a77531151.
Works local and remote (running live TrenchBoot). I changed `GNU GRUB` in grub menu to some other text and after deployment it changed and system still booted. I tried to boot the same image on UEFI and it worked okay and `GNU GRUB` was unchanged.
Now the differences:

* After deploying `grub` when booting there are additional errors:
  ```
  Jul 15 15:46:49 tb systemd[1]: Failed to start Network Name Resolution.
  Jul 15 15:46:49 tb systemd[1]: Failed to start Network Time Synchronization.
  Jul 15 15:46:49 tb systemd[1]: Failed to start User Login Management.
  ```
* After login to root there is additional output on console, possibly related to failed `User Login Management`
  ```
  Jul 15 15:49:27 tb authpriv.notice login[342]: ROOT LOGIN  on '/dev/ttyS0'
  ```

Those changes don't appear when I deploy whole image `tb-minimal-image` in which I also update legacy-grub after copying/rsyncing whole rootfs & boot.

Currently updating grub can be summed up as (on QEMU remote):
* rsync rootfs changes made by this recipe
* ```shell
   grub-mkimage -p '(hd0,msdos1)/grub' -d /usr/lib/grub/i386-pc \
     -o /usr/lib/grub/i386-pc/core.img -O i386-pc at_keyboard biosdisk boot \
     chain configfile ext2 fat linux ls part_msdos reboot serial vga
  ```
* ```shell
  grub-bios-setup -v --device-map=<(echo "(hd0) /dev/vda") -r 'hd0,msdos1' \
    -d /usr/lib/grub/i386-pc /dev/vda
  ```